api-service/src/controllers/DatasetRead/DatasetRead.ts

```diff
@@ -34,7 +35,14 @@ const datasetRead = async (req: Request, res: Response) => {
     const { fields, mode } = req.query;
     const attributes = !fields ? defaultFields : _.split(<string>fields, ",");
     const dataset = (mode == "edit") ? await readDraftDataset(dataset_id, attributes) : await readDataset(dataset_id, attributes)
-    if(!dataset) {
+    if (dataset.connectors_config) {
+        dataset.connectors_config = dataset.connectors_config.map((connector: any) => ({
```

Comment:
Wouldn't this throw an exception if the dataset is undefined. Shouldn't it be in the else block before the response is sent?

Reply:
Throwing dataset not found error if dataset is null before execution of logic